bindata/network/ovn-kubernetes-ipsec/ovn-ipsec.yaml

```diff
@@ -0,0 +1,207 @@
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: ovn-ipsec
+  namespace: openshift-ovn-kubernetes
+  annotations:
+    kubernetes.io/description: |
+      This DaemonSet launches the ovn ipsec networking components for all nodes.
+    release.openshift.io/version: "{{.ReleaseVersion}}"
+spec:
+  selector:
+    matchLabels:
+      app: ovn-ipsec
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        app: ovn-ipsec
+        component: network
+        type: infra
+        openshift.io/component: network
+        kubernetes.io/os: "linux"
+    spec:
+      serviceAccountName: ovn-kubernetes-node
+      hostNetwork: true
+      hostPID: true
+      priorityClassName: "system-node-critical"
+      # volumes in all containers:
+      # (container) -> (host)
+      # /etc/openvswitch -> /var/lib/openvswitch/etc - ovsdb system id
+      # /var/lib/openvswitch -> /var/lib/openvswitch/data - ovsdb data
+      # /run/openvswitch -> tmpfs - ovsdb sockets
+      # /env -> configmap env-overrides - debug overrides
+      containers:
+      # ovs-monitor-ipsec and libreswan daemons
+      - name: ovn-ipsec
+        image: "{{.OvnImage}}"
+        command:
+        - /bin/bash
+        - -c
+        - |
+          #!/bin/bash
+
+          function cleanup()
+          {
+            # Stop ovn-monitor-ipsec
+            /usr/share/openvswitch/scripts/ovs-ctl stop-ovs-ipsec
+            # Stop libreswan
+            /usr/libexec/ipsec/whack --shutdown
+            /sbin/ip xfrm policy flush
+            /sbin/ip xfrm state flush
+            /usr/sbin/ipsec --stopnflog
+            exit 0
+          }
+          trap cleanup SIGTERM
+
+          # Every time we restart the container, we will create a new key pair
+          echo "Configuring IPsec keys"
+
+          # We use the system-id as the CN. This is a requirement for OVN
+          cn=$(ovs-vsctl --retry -t 60 get Open_vSwitch . external-ids:system-id | tr -d "\"")
+
+          mkdir -p /etc/openvswitch/keys
+          pushd /etc/openvswitch/keys
+          umask 077 && openssl genrsa -out ipsec-privkey.pem 2048
+          openssl req -new -text \
```

Comment:
So, this generates a key and submits a CSR. That's fine... but this needs to be a bit more of an idempotent reconciliation loop that

1. Looks to see if a signed certificate exists on disk that is not about to expire
2. Generates a key + CSR if needed
3. Configures OVN-k
4. GOTO 1.

Reply:
You might have to write this in go.

Reply:
Okay, if certificate lifecycle isn't going to be solveable yet, then at least let's shortcut and not regen the certificate if it's not too old.

Reply:
Why should we worry about not regenerating the certificate? It simplifies things a lot if we dont check validity,etc. When we include better support of certificate rotation, we can refactor and maybe write it in go.

Reply:
It's not necessary, but it does mean fewer things could go wrong on startup. If you're confident that cert rotations are foolproof, then you can ignore this suggestion. If there's a chance they cause issues - the fewer we do, the better :-)

Reply:
Thats not a bet I want to take :) I will check for cert end-date validity. Apart from anything else it will speed things up significantly.